test/dtm.test.js

```diff
@@ -40,45 +40,218 @@
+    });
+
+    it('Coded phy LE is supported on nRF52840', async () => {
+        expect(await dtm.setupPhy(DTM.DTM.DTM_PARAMETER.PHY_LE_CODED_S8)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setupPhy(DTM.DTM.DTM_PARAMETER.PHY_LE_CODED_S2)).toEqual(Buffer.from(['0x00', '0x00']));
     });
 
-    it(`Transmitter command should be ['0x80', '0x00']`, () => {
-        expect(dtm.createTransmitterCMD()).toEqual(Buffer.from(['0x80', '0x00']));
+    it('Setup modulation index', async () => {
+        expect(await dtm.setupModulation(DTM.DTM.DTM_PARAMETER.STANDARD_MODULATION_INDEX)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setupModulation(DTM.DTM.DTM_PARAMETER.STABLE_MODULATION_INDEX)).toEqual(Buffer.from(['0x00', '0x00']));
     });
 
-    it(`Receiver command should be ['0x40', '0x00']`, () => {
-        expect(dtm.createReceiverCMD()).toEqual(Buffer.from(['0x40', '0x00']));
+    it('Read features', async () => {
+        const readFeatures = async () => {
+            const buffer = await dtm.setupReadFeatures();
+            buffer[0] &= 0x01; // LE_Test_Status_Event
+            buffer[1] &= 0x80; // Success
+            return buffer;
+        };
+        expect(await readFeatures()).toEqual(Buffer.from(['0x00', '0x00']));
     });
 
-    it(`End command hould be ['0xC0', '0x00']`, () => {
-        expect(dtm.createEndCMD()).toEqual(Buffer.from(['0xc0', '0x00']));
+    it('Set TX power', async () => {
+        expect(await dtm.setupReset()).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(-40)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(-20)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(-16)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(-12)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(-8)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(-4)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(0)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(2)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(3)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(4)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(5)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(6)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(7)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(8)).toEqual(Buffer.from(['0x00', '0x00']));
+        expect(await dtm.setTxPower(9)).toEqual(Buffer.from(['0x00', '0x01']));
+        expect(await dtm.setTxPower(1)).toEqual(Buffer.from(['0x00', '0x01']));
+        expect(await dtm.setTxPower(-41)).toEqual(Buffer.from(['0x00', '0x01']));
     });
+
+/* This test will make other test fail.
```

Comment:
Please remove commented out code, or if this will be useful at a later time in this case you can use `it.skip()`.